src/UglyToad.PdfPig/Export/AltoXmlTextExporter.cs

```diff
@@ -0,0 +1,5337 @@
+            };
+
+            var fileIdentifier = new AltoDocument.AltoFileIdentifier()
+            {
+                FileIdentifierLocation = null,
+                Value = null
+            };
+
+            return new AltoDocument.AltoDescription()
+            {
+                MeasurementUnit = AltoDocument.AltoMeasurementUnit.Pixel, // need to check that
+                Processings = new[] { processing },
+                SourceImageInformation = new AltoDocument.AltoSourceImageInformation()
+                {
+                    DocumentIdentifiers = new AltoDocument.AltoDocumentIdentifier[] { documentIdentifier },
+                    FileIdentifiers = new AltoDocument.AltoFileIdentifier[] { fileIdentifier },
+                    FileName = fileName
+                }
+            };
+        }
+
+        private string Serialize(AltoDocument altoDocument)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(AltoDocument));
+            var settings = new XmlWriterSettings()
+            {
+                //Encoding = new System.Text.UTF8Encoding(true),
+                Indent = true,
+                IndentChars = indentChar,
+                OmitXmlDeclaration = true // hack to manually handle utf-8
+            };
+
+            using (var stringWriter = new System.IO.StringWriter())
+            using (var xmlWriter = XmlWriter.Create(stringWriter, settings))
+            {
+                stringWriter.WriteLine("<?xml version=\"1.0\" encoding=\"UTF-8\"?>"); // hack to manually handle utf-8
+                serializer.Serialize(xmlWriter, altoDocument);
+                return stringWriter.ToString();
+            }
+        }
+
+        private static AltoDocument Deserialize(string xmlPath)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(AltoDocument));
+
+            using (var reader = XmlReader.Create(xmlPath))
+            {
+                return (AltoDocument)serializer.Deserialize(reader);
+            }
+        }
+    }
+
+    #region ALTO Schema
+    /********************************************************************************
+     * Alto version 4.1 https://github.com/altoxml/schema/blob/master/v4/alto-4-1.xsd
+     * Auto-generated by xsd and improved by BobLD
+     ********************************************************************************/
+
+    /// <summary>
+    /// [Alto] Alto Schema root
+    /// <para>Version 4.1</para>
+    /// See https://github.com/altoxml/schema
+    /// </summary>
+    [EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+    [GeneratedCodeAttribute("xsd", "4.6.1055.0")]
+    [SerializableAttribute()]
+    [DebuggerStepThroughAttribute()]
+    [DesignerCategoryAttribute("code")]
+    [XmlTypeAttribute(Namespace = "http://www.loc.gov/standards/alto/ns-v4#")]
+    [XmlRootAttribute("alto", Namespace = "http://www.loc.gov/standards/alto/ns-v4#", IsNullable = false)]
+    public class AltoDocument
```

Comment:
I'd move this to a new file and for each of the properties (except where required by the `isXXFieldSpecified` setters) change it to be an auto property, I'll add an example of this below...